include/materials/modified_cam_clay.h

```diff
@@ -127,15 +125,31 @@ class ModifiedCamClay : public Material<Tdim> {
   using Material<Tdim>::console_;
 
  private:
-  //! Elastic stiffness matrix
-  Matrix6x6 de_;
-  //! Plastic stiffness matrix
-  Matrix6x6 dp_;
+  //! Compute elastic tensor
+  //! \param[in] stress Stress
+  //! \param[in] state_vars History-dependent state variables
+  Eigen::Matrix<double, 6, 6> compute_elastic_tensor(
+      const Vector6d& stress, mpm::dense_map* state_vars);
+
+  //! Compute constitutive relations matrix for elasto-plastic material
+  //! \param[in] stress Stress
+  //! \param[in] dstrain Strain
+  //! \param[in] particle Constant point to particle base
+  //! \param[in] state_vars History-dependent state variables
+  //! \param[in] hardening Boolean to consider hardening, default=true. If
+  //! perfect-plastic tensor is needed pass false
+  //! \retval dmatrix Constitutive relations mattrix
+  Matrix6x6 compute_elasto_plastic_tensor(const Vector6d& stress,
+                                          const Vector6d& dstrain,
+                                          const ParticleBase<Tdim>* ptr,
+                                          mpm::dense_map* state_vars,
+                                          bool hardening = true) override;
+
   //! General parameters
   //! Density
   double density_{std::numeric_limits<double>::max()};
-  //! Youngs modulus
-  double youngs_modulus_{std::numeric_limits<double>::max()};
+  //! Shear Modulus
```

Comment:
Why do you use shear modulus instead of Young's modulus?

Reply:
No specific reasons. In the modified Cam-Clay, the input value of the shear modulus or Young's modulus is just used for the first time step. Particularly, if the mean pressure is zero/negative. This is crucial if initial stresses are zero in all directions. The reason why E is replaced by G is that it is normally easier to calculate G from triaxial lab testing. So maybe this is better. I also see many papers use this instead of E. Any thoughts?

Reply:
Ok, just for curiosity. Btw, as far as I know, the triaxial test provides E as the secant slope of the axial stress-axial strain curve, and G is converted from E assuming a particular value of Poisson's ratio. Torsional shear test directly provides G.

Reply:
Actually, you might be right. In triaxial lab data, one usually gets the E50, instead of G. I was thinking G might be easier since one normally plots q vs dev_strain too. At this stage, I don't have much preference. But I hope the user should be able to compute them interchangeably using Lame's transformation.

Reply:
Thank you for the reply. Besides the above aspects, I don't have any preference for the code. Using G as an input variable is also ok for me.